ambry-cloud/src/main/java/com/github/ambry/cloud/azure/StorageClient.java

```diff
@@ -423,10 +424,20 @@ protected abstract BlobServiceAsyncClient buildBlobServiceAsyncClient(HttpClient
    * @param throwable associated with this error.
    */
   private void onError(Throwable throwable) {
-    logger.error("Blob store operation failed due to exception: " + throwable.toString());
-    azureMetrics.storageClientOperationExceptionCount.inc();
-    // All retries must have been completed internally by now.
-    azureMetrics.storageClientFailureAfterRetryCount.inc();
+    boolean logError = true;
+    if (throwable instanceof BlobStorageException) {
+      BlobStorageException bse = (BlobStorageException) throwable;
+      // Do not log this error if status = 409 and blob-exists, it's just noise
+      logError = !(bse.getStatusCode() == BLOB_ALREADY_EXISTS && bse.getMessage().equals("The specified blob already exists"));
+    }
+    if (logError) {
+      logger.error("Blob store operation failed due to exception: " + throwable.toString());
+      azureMetrics.storageClientOperationExceptionCount.inc();
+      // All retries must have been completed internally by now.
+      azureMetrics.storageClientFailureAfterRetryCount.inc();
+    } else {
+      logger.debug("Blob store operation failed due to exception: " + throwable);
```

Comment:
throwable.toString()

Reply:
Yeah, intellij recommended its not necessary.